lib/iamspec/matcher/be_allowed_to.rb

```diff
@@ -1,29 +1,21 @@
+require_relative 'generic_allowed_to.rb'
+
 def be_allowed_to(action)
   BeAllowedTo.new(action)
 end
 
-class BeAllowedTo
+class BeAllowedTo < GenericAllowedTo
   def initialize(action)
     @action = action
   end
 
   def matches?(subject)
-    @subject = subject
-    iam  = Aws::IAM::Client.new(region: 'us-east-1')
-    resp = iam.simulate_principal_policy({
-        policy_source_arn: subject.arn,
-        action_names:      @action.action_names,
-        resource_arns:     @action.resource_arns,
-    })
-
-    # puts "fooo: #{resp.evaluation_results}"
-
-    @evaluation_results = resp.evaluation_results
+    @evaluation_results = super(subject)
     failure_results(@evaluation_results).empty?
   end
 
   def description
-    "be allowed to #{@action.to_s}"
+    "be allowed to #{@action.to_s} #{@action.userid ? 'with userid ' + @action.userid : ''}"
```

Comment:
This adds a whitespace to the description in case `@action.userid` is not set, breaking a unit test: 
```
  1) be_allowed_to matcher all cases description should have a nice description
     Failure/Error: expect(subject.description).to eq('be allowed to SomeAction')

       expected: "be allowed to SomeAction"
            got: "be allowed to SomeAction "

       (compared using ==)
     # ./spec/matcher_be_allowed_to_spec.rb:9:in `block (4 levels) in <top (required)>'

```

Reply:
Fixed